packages/driver/src/rawConn.ts

```diff
@@ -152,7 +152,7 @@ export class RawConnection extends BaseRawConnection {
       this.buffer.takeMessage() &&
       this.buffer.getMessageType() === chars.$E
     ) {
-      newErr.source = this._parseErrorMessage();
+      newErr.cause = this._parseErrorMessage();
```

Comment:
Does this logic truly has to be done after the waiter check above? If not it could be merged into error constructor, otherwise I think it's the only remaining assignment.

Reply:
I don't think it makes sense to refactor this logic in the context of this PR, so I think it's safest to leave this one as a conditional assignment.